src/OrchardCore.Modules/Orchard.Users/Services/UserService.cs

```diff
@@ -49,55 +53,83 @@ public UserService(UserManager<User> userManager, IOptions<IdentityOptions> iden
-                            reportError("Password", T["Passwords must have at least one lowercase ('a'-'z')."]);
-                            break;
-                        case "PasswordRequiresUpper":
-                            reportError("Password", T["Passwords must have at least one uppercase('A'-'Z')."]);
-                            break;
-                        case "PasswordRequiresNonAlphanumeric":
-                            reportError("Password", T["Passwords must have at least one non letter or digit character."]);
-                            break;
-                        case "PasswordTooShort":
-                            reportError("Password", T["Passwords must be at least {0} characters.", _identityOptions.Value.Password.RequiredLength]);
-                            break;
+                ProcessIdentityErrors(identityResult.Errors, user, reportError);
+            }
+
+            return identityResult.Succeeded;
+        }
 
-                        // Password confirmation
-                        case "PasswordMismatch":
-                            reportError("PasswordConfirmation", T["Incorrect password."]);
-                            break;
 
-                        // User name
-                        case "InvalidUserName":
-                            reportError("UserName", T["User name '{0}' is invalid, can only contain letters or digits.", user.UserName]);
-                            break;
-                        case "DuplicateUserName":
-                            reportError("UserName", T["User name '{0}' is already used.", user.UserName]);
-                            break;
+        public async Task<bool> ChangePasswordAsync(User user, string currentPassword, string newPassword, Action<string, string> reportError)
+        {
+            var identityResult = await _userManager.ChangePasswordAsync(user, currentPassword, newPassword);
 
-                        // Email
-                        case "InvalidEmail":
-                            reportError("Email", T["Email '{0}' is invalid.", user.Email]);
-                            break;
-                        default:
-                            reportError(string.Empty, T["Unexpected error: '{0}'.", error.Code]);
-                            break;
-                    }
-                }
+            if(!identityResult.Succeeded)
+            {
+                ProcessIdentityErrors(identityResult.Errors, user, reportError);
             }
 
             return identityResult.Succeeded;
         }
+
+        public async Task<User> GetAuthenticatedUserAsync(ClaimsPrincipal principal)
+        {
+            if(principal == null)
+            {
+                return null;
+            }
+
+            return await _userManager.GetUserAsync(principal);
+        }
+
+        private void ProcessIdentityErrors(IEnumerable<IdentityError> errors, User user, Action<string, string> reportError)
```

Comment:
I am not a native speaker, but the name `ValidateUser` gives me impression that the method actually preforms some validation of the `user` parameter. In fact the validation is done by the `UserManager` class. What about `ProcessValidationErrors`?

Of course, if I am wrong, I will change the name to `ValidateUser` ...